cpp/fury/util/string_util.h

```diff
@@ -0,0 +1,33 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+#pragma once
+
+#include <string>
+
+namespace fury {
+
+bool isLatin_Baseline(const std::string &str);
+bool isLatin_AVX2(const std::string &str);
+bool isLatin_SSE2(const std::string &str);
+bool isLatin_NEON(const std::string &str);
+bool isLatin_RISCV(const std::string &str);
+std::string generateRandomString(size_t length);
```

Comment:
Could we add a method which select implementation based the macro. Otherwise, how do know which method should be invoked?

Reply:
I get it